pep-0708.rst

```diff
@@ -0,0 +1,849 @@
+provides files for that project is another workable solution that is safe if the
+order is specified correctly.
+
+However, this has been rejected for a number of reasons:
+
+- We've spent 15+ years educating users that the ordering of repositories being
+  specified is not meaningful, and they effectively have an undefined order. It
+  would be difficult to backpedal on that and start saying that now order
+  matters.
+- Users can easily rearrange the order that they specify their repositories in
+  within a single location, but when loading repositories from multiple
+  locations (env var, conf file, requirements file, cli arguments) the order is
+  hard coded into pip. While it would be a deterministic and documented order,
+  there's no reason to assume it's the order that the user wants their
+  repositories to be defined in, forcing them to contort thow they configure pip
+  so that the implicit ordering ends up being the correct one.
+- The above can be mitigated by providing a way to explicitly declare the order
+  rather than by implicitly using the order they were defined in, however that
+  then means that the protections are not provided unless the user does some
+  explicit configuration.
+- Ordering assumes that there is a linear ordering of repositories where two
+  repositories have the same name, we always prefer repositories in the same
+  order for every project, however that is not necessarily true.
+- Relying on ordering is subtle, if I look at an ordering of repositories, I
+  have no way of knowing in advance or ensuring in advance what names are going
+  to come from what repositories. I can only know in that moment what names are
+  provided by which repositories.
+- Relying on ordering is fragile. There's no reason to assume that two disparate
+  repositories are not going to have random naming collisions, what happens if
+  I'm using a library from a lower priority repository and then a higher
+  priority repository happens to start having a colliding name.
+- In cases where ordering does the wrong thing, it does so silently, with no
+  feedback given to the user. This is by design because it doesn't actually know
+  what the wrong or right thing is, it's just hoping that order will give the
+  right thing, and if it does then users are protected without any breakage.
+  However, when it does the wrong thing, users are left with a very confusing
+  behavior coming from pip, where it's just silently installing the wrong thing.
+
+There is a variant of this idea which effectively says that it's really just
+PyPI's nature of open registration that causes the real problems, so if we treat
+all repositories but the "default" one as equal priority, and then treat the
+default one as a lower priority then we'll fix things.
+
+That is true in that it does improve things, but it has many of the same
+problems as the general ordering idea (though not all of them), but it comes
+along with some of it's own:
```

Comment:
```suggestion
along with one of its own:
```

* Fix it's -> its
* There's only one thing listed here (which doesn't make so much sense to put in a list anyway...)

Did you mean to include something else?

Reply:
I think I originally had 2 items, then decided the second didn't make sense and dropped it and never fixed this.